pkgs/development/libraries/dnnl/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, lib, fetchFromGitHub, substituteAll, autoPatchelfHook
+, cmake, bash
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dnnl";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "mkl-dnn";
+    rev = "v${version}";
+    sha256 = "150cdyfiw4izvzmbmdqidwadppb1qjmzhpaqjczm397ygi1m92l1";
+  };
+
+  patches = [ (substituteAll {
+    src = ./bash-to-sh.patch;
+    inherit bash;
+  }) ];
+
+  outputs = [ "out" "dev" "doc" ];
+
+  nativeBuildInputs = [ cmake ];
+  checkInputs = [ autoPatchelfHook ];
+
+  # The test executables aren't able to find libmkldnn.so through
+  # LD_LIBRARY_PATH, so patch them.
+  doCheck = true;
+  preCheck = ''
+    addAutoPatchelfSearchPath $PWD/src
+    autoPatchelf tests examples
```

Comment:
I think something like:
```
  export LD_LIBRARY_PATH=$PWD/src/build
```
would be more appropriate and drop the autopatchelf

generally autopatchelf is to help with precompilied binaries, not test binaries

Reply:
I started digging into it, but I wasn't able to get this to work with `LD_LIBRARY_PATH`. It looks like the test driver re-launches with a subprocess or something that cleans out the var (???) and it doesn't get caught, hence the original comment by @alexarice about the `LD_LIBRARY_PATH` env var not working.  

I did upstream the patch on the shell to Intel, who says they'll release it in 1.2.0, so at least we can clean that part of the packaging up eventually.

@GrahamcOfBorg build dnnl

Reply:
Actually, I'm a liar: the tests do build, run, and pass just fine without the autopatchelf hook and just
```nix
  preCheck = ''
    export LD_LIBRARY_PATH=$PWD/src
  '';
```

but we then later run into issues at installation and dependency resolution time . . . 

```
checking for references to /build/ in /nix/store/3g8jhc6iwipb22a03grlif5y6bddb09j-dnnl-1.1.2-doc...
automatically fixing dependencies for ELF files
searching for dependencies of /nix/store/yixg9d3dfryiypffdn90zl30z6kk0yz7-dnnl-1.1.2/lib/libdnnl.so.1.1
  libstdc++.so.6 -> not found!
  libgomp.so.1 -> not found!
builder for '/nix/store/11bqvbfw2a3sxjpyfg3fycva2mxm408i-dnnl-1.1.2.drv' failed with exit code 1
error: build of '/nix/store/11bqvbfw2a3sxjpyfg3fycva2mxm408i-dnnl-1.1.2.drv' failed
```

Since we actually depend on the patchelf hook for mutating the installed binaries themselves, I will update the comment to accurately describe the state of the world!

Reply:
if that's the cause, then do the autoPatchelfHook, but let it do it's thing during the  fixup phase instead of the installCheck

Reply:
The claim of being a lie was itself a lie, due to an unrelated issue.  This takes a while to rebuild and re-run the tests on my laptop, so by the time it finished with a failure I had forgotten what I was attempting :)

The tests actually pass just fine with a simple `LD_LIBRARY_PATH` env var set, and the installed artifacts look good, so I think what we've got right now in the PR is the optimal fix in terms of both simplicity for test running, clarity of maintenance, and avoiding pollution of the installed artifacts for fixing test issues.